src/cache.rs

```diff
@@ -96,8 +96,19 @@ pub struct Cache {
     name: &'static str,
 
     /// Directory to use for storing cache items. Will be created if it does not exist.
+    ///
+    /// Leaving this as None will disable this cache.
     cache_dir: Option<PathBuf>,
 
+    /// Directory to use for temporary files.
+    ///
+    /// When writing a new file into the cache it is best to write it to a temporary file in
+    /// a sibling directory, once fully written it can then be atomically moved to the
+    /// actual location withing the `cache_dir`.
+    ///
+    /// Just like for `cache_dir` when this cache is disabled this will be `None`.
+    tmp_dir: Option<PathBuf>,
```

Comment:
Is there a situation where one wants to make this configurable? afaict it is not exposed as config param but I think we should avoid exposing it entirely.. I think `NamedTempFile::new()` is never what you want

Reply:
My intention was also not to expose this.  You only get to chose a cache directory (or not) and we decide what to do inside of there.  It makes me a little sad that you can create a `Cache` with `cache_dir=Some(...)` and `tmp_dir=None` or the other way around.  But only code in this module should be able to do that.